filesystem/path.h

```diff
@@ -184,49 +206,79 @@ class path {
 
     std::string str(path_type type = native_path) const {
         std::ostringstream oss;
+        bool ends_slash = false;
 
-        if (m_type == posix_path && m_absolute)
-            oss << "/";
+        if (m_type == windows_path && m_volume != 0)
+            oss << m_volume << ':';
+
+        if (m_absolute || m_starts_slash) {
+            oss << slash();
+            ends_slash = true;
+        }
 
-        for (size_t i=0; i<m_path.size(); ++i) {
+        for (size_t i = 0; i < m_path.size(); ++i) {
```

Comment:
```c++
for (size_t i = 0, path_size = m_path.size(); i < path_size; ++i) {
```

Probably a good idea to cache the path size here for posterity so you're not querying the size of the path each iteration. Can't guarantee the call will be optimized down to a simple lookup on certain platforms.